### ISSUE TITLE
url_detect()?

### ISSUE THREAD

**Issue by hrbrmstr** [Ironholds/urltools#37]
vis a vis http://stackoverflow.com/questions/33773299/r-regex-to-extract-url-from-text

**Comment by Ironholds** [comment]
Ooh, painful. We'd have to do it in R for /that/ use case. We could do it in C++, if we wanted _actual_ URLs, but those aren't actual URLs.

**Comment by hrbrmstr** [comment]
somewhat painful. If we go for "best guess regex" then we can use gruber's fairly-comprehensive-but-not-too-aggressive regex for it https://gist.github.com/gruber/249502 and https://gist.github.com/gruber/8891611

**Comment by Ironholds** [comment]
Good idea! We could always have url_detect(x, strict = FALSE), where strict determines if it relies on a :// find (easy enough in C++, and fast) or a regex (harder and slow)

**Comment by Ironholds** [comment]
Could we use stringi for this?

**Comment by Ironholds** [comment]
Note to self: we can use C++11 for this, since everyone and their pet dog seems to be into C++11 these days.

**Comment by hrbrmstr** [comment]
FWIW our pet cat is also into C++11.
